In [1]:
import pandas as pd

df = pd.read_csv("data.csv")
df.head()

,ad_id,reporting_start,reporting_end,campaign_id,fb_campaign_id,age,gender,interest1,interest2,interest3,impressions,clicks,spent,total_conversion,approved_conversion
0,708746,17/08/2017,17/08/2017,916,103916,30-34,M,15,17,17,7350.0,1,1.43,2.0,1.0
1,708749,17/08/2017,17/08/2017,916,103917,30-34,M,16,19,21,17861.0,2,1.82,2.0,0.0
2,708771,17/08/2017,17/08/2017,916,103920,30-34,M,20,25,22,693.0,0,0.00,1.0,0.0
3,708815,30/08/2017,30/08/2017,916,103928,30-34,M,28,32,32,4259.0,1,1.25,1.0,0.0
4,708818,17/08/2017,17/08/2017,916,103928,30-34,M,28,33,32,4133.0,1,1.29,1.0,1.0


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1143 entries, 0 to 1142
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ad_id                1143 non-null   int64  
 1   reporting_start      1143 non-null   str    
 2   reporting_end        1143 non-null   str    
 3   campaign_id          1143 non-null   str    
 4   fb_campaign_id       1143 non-null   str    
 5   age                  1143 non-null   str    
 6   gender               1143 non-null   str    
 7   interest1            1143 non-null   int64  
 8   interest2            1143 non-null   int64  
 9   interest3            1143 non-null   int64  
 10  impressions          1143 non-null   float64
 11  clicks               1143 non-null   int64  
 12  spent                1143 non-null   float64
 13  total_conversion     761 non-null    float64
 14  approved_conversion  761 non-null    float64
dtypes: float64(4), int64(5), str(6)
memory usage: 134

In [3]:
df = df[df['approved_conversion'].notna()]

df['ROAS'] = df['approved_conversion'] / df['spent']

In [4]:
df[['campaign_id', 'spent', 'approved_conversion', 'ROAS']].head()

,campaign_id,spent,approved_conversion,ROAS
0,916,1.43,1.0,0.699301
1,916,1.82,0.0,0.000000
2,916,0.00,0.0,NaN
3,916,1.25,0.0,0.000000
4,916,1.29,1.0,0.775194


# ROAS Calculation

Return on Ad Spend (ROAS) was calculated by dividing approved conversions by total ad spend. Since direct revenue data was not available, approved conversions were used as a proxy for revenue.

In [5]:
campaign_roas = df.groupby('campaign_id').agg({
    'spent': 'sum',
    'approved_conversion': 'sum'
}).reset_index()

campaign_roas['ROAS'] = campaign_roas['approved_conversion'] / campaign_roas['spent']

campaign_roas

,campaign_id,spent,approved_conversion,ROAS
0,1178,16577.159998,378.0,0.022802
1,916,149.710001,24.0,0.160310
2,936,2893.369999,183.0,0.063248


In [6]:
campaign_roas = campaign_roas.sort_values(by='ROAS', ascending=False)
campaign_roas

,campaign_id,spent,approved_conversion,ROAS
1,916,149.710001,24.0,0.160310
2,936,2893.369999,183.0,0.063248
0,1178,16577.159998,378.0,0.022802


# Campaign Ranking

Campaigns were ranked based on ROAS to identify the most and least efficient marketing investments.

# Campaign Performance Analysis

Campaign 916 shows the highest ROAS despite having the lowest spend, indicating efficient use of budget. Campaign 936 has moderate performance, while Campaign 1178 has the lowest ROAS despite the highest spend, suggesting inefficient spending.

# Scaling Recommendations

Scale Up – Campaign 916:

This campaign has the highest ROAS, meaning it generates the most conversions per unit of spend. Increasing budget here is likely to improve overall returns.

Scale Down – Campaign 1178:

Despite high spending, this campaign delivers the lowest ROAS. This indicates inefficient budget usage, and reducing spend or optimizing this campaign would improve overall performance.